CMakeLists.txt

```diff
@@ -7,7 +7,7 @@ else()
   cmake_minimum_required(VERSION 3.5.1)
 endif()
 
-project(dart)
+project(DART)
```

Comment:
This may impact on many places that use `${PROJECT_NAME}` such as the target names and the component names, which shall breaks the backward compatibility in finding components.

We had the [consensus](https://github.com/dartsim/dart/issues/1198#issuecomment-464569220) on changing the project name to `dartsim`, but that's for the next major release.

I think the most minimal changes to address this issue without breaking the backward compatibility would be using the lowercased project name to everywhere used `${PROJECT_NAME}` by doing something like:
```
string(TOLOWER ${PROJECT_NAME} PROJECT_NAME_LOWERCASE)
```
similar to [`PROJECT_NAME_UPPERCASE`](https://github.com/dartsim/dart/blob/15b5af8a3ed46a8c5350f0d01e767ff90b14eb2e/CMakeLists.txt#L12).

Reply:
I think the main thing is that colcon wants the project name (specifically the text inside `project()`) to match the name passed to `find_package()`. So if you use `project(dartsim)`, then I think colcon would want to see `find_package(dartsim ...` in order to infer the dependency properly.

Reply:
Yeah, I meant to chaning them all to use `dartsim` instead of `dart` (or `DART`), but again it's for the next major release.